metadata-ingestion/src/datahub/ingestion/source/azure_data_factory/adf_models.py

```diff
@@ -0,0 +1,622 @@
+"""Pydantic models for Azure Data Factory API responses.
+
+These models provide type safety and validation for ADF REST API responses.
+Field names match the Azure API response structure (camelCase).
+
+API Documentation: https://learn.microsoft.com/en-us/rest/api/datafactory/
+"""
+
+from datetime import datetime
+from typing import Any, Optional, Union
+
+from pydantic import BaseModel, ConfigDict, Field, model_validator
+from typing_extensions import TypedDict
+
+# Type aliases for common JSON value types in ADF API responses
+# Azure API parameters and variables can contain primitive types
+JsonPrimitive = Union[str, int, float, bool, None]
+
+
+# TypedDict for well-known structures in ADF API responses
```

Comment:
we are rarely using `TypedDict`, any reason why being used here?
(asking in case we should use them more frequently 😅 )